composer/trainer/trainer.py

```diff
@@ -2066,9 +2078,18 @@ def _accumulate_time_across_ranks(
         """
         # Samples and tokens should be summed
         # Batch time should be the value from rank 0
-        sample_token_tensor = self.state.device.tensor_to_device(
-            torch.tensor([num_samples, num_tokens], dtype=torch.int))
+        if isinstance(num_samples, float):
+            sample_token_tensor = self.state.device.tensor_to_device(
+                torch.tensor([num_samples, num_tokens], dtype=torch.float32))
+        else:
+            sample_token_tensor = self.state.device.tensor_to_device(
+                torch.tensor([num_samples, num_tokens], dtype=torch.int))
         dist.all_reduce(sample_token_tensor, reduce_operation='SUM')
+        if isinstance(num_samples, float):
+            sample_token_tensor_int = sample_token_tensor.to(torch.int)
+            if torch.any(torch.ne(sample_token_tensor_int, sample_token_tensor)):
+                raise ValueError('The sums of samples and tokens across ranks should each be integers.')
+            sample_token_tensor = sample_token_tensor_int
```

Comment:
Can you put a comment describing the float scenarios here please?

Reply:
Done, [here](https://github.com/mosaicml/composer/blob/541913d19cd7f3ebfe81add0fd6b9fc9410ab166/composer/trainer/trainer.py#L2082).